pkg/codegen/python/gen_test.go

```diff
@@ -59,164 +56,160 @@ func TestRelPathToRelImport(t *testing.T) {
 }
 
 func TestGeneratePackage(t *testing.T) {
+	// To speed up these tests, we will generate one common
+	// virtual environment for all of them to run in, rather than
+	// having one per test.
+	err := buildVirtualEnv()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
 	test.TestSDKCodegen(t, &test.SDKCodegenOptions{
 		Language:   "python",
 		GenPackage: GeneratePackage,
 		Checks: map[string]test.CodegenCheck{
 			"python/py_compile": pyCompileCheck,
+			"python/test":       pyTestCheck,
 		},
 	})
 }
 
-// Checks generated code for syntax errors with `python -m compile`.
-func pyCompileCheck(t *testing.T, codeDir string) {
-	ex, _, err := python.CommandPath()
-	require.NoError(t, err)
-	cmdOptions := integration.ProgramTestOptions{}
-	err = filepath.Walk(codeDir, func(path string, info filesystem.FileInfo, err error) error {
-		require.NoError(t, err) // an error in the walk
-		if info.Mode().IsRegular() && strings.HasSuffix(info.Name(), ".py") {
-			path, err = filepath.Abs(path)
-			require.NoError(t, err)
-			err = integration.RunCommand(t, "python syntax check",
-				[]string{ex, "-m", "py_compile", path}, codeDir, &cmdOptions)
-			require.NoError(t, err)
-		}
-		return nil
-	})
-	require.NoError(t, err)
+func absTestsPath() (string, error) {
+	hereDir, err := filepath.Abs(".")
+	if err != nil {
+		return "", err
+	}
+	return hereDir, nil
 }
 
-func TestGenerateTypeNames(t *testing.T) {
-	test.TestTypeNameCodegen(t, "python", func(pkg *schema.Package) test.TypeNameGeneratorFunc {
-		// Decode python-specific info
-		err := pkg.ImportLanguages(map[string]schema.Language{"python": Importer})
-		require.NoError(t, err)
+func virtualEnvPath() (string, error) {
+	hereDir, err := absTestsPath()
+	if err != nil {
+		return "", err
+	}
+	return filepath.Join(hereDir, "venv"), nil
+}
 
-		info, _ := pkg.Language["python"].(PackageInfo)
+// To serialize shared `venv` operations; without the lock running
+// tests with `-parallel` causes sproadic failure.
+var venvMutex *sync.Mutex = &sync.Mutex{}
```

Comment:
Can't do `pip install -e .` in parallel on a shared venv, withohut a mutex. Unfortunate.